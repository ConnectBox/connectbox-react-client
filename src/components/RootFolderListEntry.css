.home-icon-container {
    width: 65px;
    height: 70px;
}

.home-icon-container img {
  display: block;
  width: 100%;
  height: auto;
}

.centered {
    text-align: center;
}

table.collapsed span.row_name {
    display: none;
}

table span.directory {
    font-weight: bold;
}
