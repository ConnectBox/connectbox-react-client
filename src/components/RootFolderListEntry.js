import './RootFolderListEntry.css'

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {  topLevelCategory } from '../utils'
import { Link } from 'react-router-dom'

class RootFolderListEntry extends Component {
    static propTypes = {
        file: PropTypes.object.isRequired,
        contentRoot: PropTypes.string.isRequired,
        displayFolderName: PropTypes.bool.isRequired,
        iconMetadata: PropTypes.object.isRequired
    }

  render () {
    const { file, contentRoot, displayFolderName, iconMetadata } = this.props
    const { icon_file, name } = file
    const fileLink = `#/${name}`

    let spanClass = 'directory'
    if (!displayFolderName) {
      spanClass = spanClass + ' row_name'
    }

    let folderNameSpan = undefined
    if (icon_file) {
      if (displayFolderName) {
        folderNameSpan = (
          <span>{name}</span>
        )
      }

      return (
        <tr>
          <td className="centered">
            <Link to={fileLink}>
              <div className="home-icon-container">
                  <img alt="" src={`${contentRoot}/${icon_file}`} />
              </div>
              {folderNameSpan}
              <span className={spanClass}></span>
            </Link>
          </td>
        </tr>
      )
    }

    if (displayFolderName) {
      folderNameSpan = (
        <span><br />{name}</span>
      )
    }

    return (
      <tr>
        <td className="centered">
          <Link to={fileLink}>
            <i className={ `fa ${topLevelCategory(file, iconMetadata)}` }></i>
            {folderNameSpan}
            <span className={spanClass}></span>
          </Link>
        </td>
      </tr>
    )
  }
}

export default RootFolderListEntry
