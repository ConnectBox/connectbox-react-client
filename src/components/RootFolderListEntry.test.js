/* global it, expect, */
import React from 'react'
import RootFolderListEntry from './RootFolderListEntry'
import renderer from 'react-test-renderer'
import { MemoryRouter as Router } from 'react-router-dom'

it('Render RootFolderListEntry', () => {
  const iconMetadata = {ids: [], names: {}}
  const file = {type: 'file', ext: 'doc', name: 'Test File'}
  const component = renderer.create(
    <Router>
      <RootFolderListEntry
        contentRoot='/content'
        key='fooEntry0'
        file={file}
        iconMetadata={iconMetadata}
        displayFolderName />
    </Router>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
