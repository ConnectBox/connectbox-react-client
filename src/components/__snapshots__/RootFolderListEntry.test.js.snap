// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RootFolderListEntry 1`] = `
<tr>
  <td
    className="centered"
  >
    <a
      href="#/Test File"
      onClick={[Function]}
    >
      <i
        className="fa "
      />
      <span>
        <br />
        Test File
      </span>
      <span
        className="directory"
      />
    </a>
  </td>
</tr>
`;
